src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -108,6 +158,70 @@ public void updateCurrentCommandField() {
 		currentCommandField.setText(commandManager.getCurrentCommandString());
 	}
 
+	private void translateCommand(){
+		TranslateStrategy translateStrategy = new TranslateStrategy(5, 5);
+		setTransformationCommand(translateStrategy);
+	}
+
+	private void scaleCommand(){
+		ScaleStrategy scaleStrategy = new ScaleStrategy(2, 2);
+		setTransformationCommand(scaleStrategy);
+	}
+
+	private void rotateCommand(){
+		RotateStrategy rotateStrategy = new RotateStrategy(2, 2, 270);
+		setTransformationCommand(rotateStrategy);
+	}
+
+	private void flipCommand(){
+		FlipStrategy flipStrategy = new FlipStrategy();
+		setTransformationCommand(flipStrategy);
+	}
+
+	private void setTransformationCommand(TransformStrategyInterface strategy) {
+		ComplexCommand currentCommand = (ComplexCommand) commandManager.getCurrentCommand();
+
+		TransformerCommandVisitorInterface translateCommandVisitor = new TransformerCommandVisitor(strategy);
+
+		Iterator<DriverCommand> iterator = currentCommand.iterator();
+
+		while(iterator.hasNext()) {
+			iterator.next().accept(translateCommandVisitor);
+
+		}
```

Comment:
iteration should be implemented in visitICompoundCommand (Visitor implementation class)